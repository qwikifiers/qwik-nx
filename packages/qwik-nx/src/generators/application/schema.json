{
  "$schema": "http://json-schema.org/schema",
  "$id": "QwikNx",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the application.",
      "pattern": "^[a-zA-Z][^:]*$",
      "x-priority": "important"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "$default": { "$source": "argv", "index": 0 },
      "x-prompt": "Which directory do you want to create the application in?"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          }
        ],
        "default": "css"
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "strict": {
      "type": "boolean",
      "description": "Creates an application with strict mode and strict type checking.",
      "default": true
    },
    "tailwind": {
      "description": "Setup Tailwind",
      "type": "boolean",
      "default": false
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "none"
    },
    "devServerPort": {
      "type": "number",
      "description": "The port for the dev server of the app.",
      "default": 5173
    },
    "previewServerPort": {
      "type": "number",
      "description": "The port for the preview server of the app.",
      "default": 4173
    }
  },
  "required": ["directory"]
}
