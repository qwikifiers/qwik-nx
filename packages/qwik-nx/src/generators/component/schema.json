{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Component",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested)."
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          },
          {
            "value": "none",
            "label": "none"
          }
        ],
        "default": "css"
      }
    },
    "skipTests": {
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "type": "boolean"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    },
    "generateStories": {
      "description": "Create Storybook stories for the component",
      "type": "boolean"
    },
    "exportDefault": {
      "description": "Whether to use default export instead of named one for the generated component.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["name", "project"]
}
