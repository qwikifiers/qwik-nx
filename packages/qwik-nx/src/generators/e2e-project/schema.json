{
  "$schema": "http://json-schema.org/schema",
  "$id": "E2eProject",
  "title": "Create an E2E app for a Qwik app",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "The name of the frontend project to test."
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "playwright"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project", "e2eTestRunner"]
}
