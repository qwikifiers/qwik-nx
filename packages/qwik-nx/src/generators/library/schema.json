{
  "$schema": "http://json-schema.org/schema",
  "$id": "Library",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Library name.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the lib is placed."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library.",
      "x-priority": "important"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          },
          {
            "value": "none",
            "label": "none"
          }
        ],
        "default": "css"
      }
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "importPath": {
      "type": "string",
      "description": "The library name used to import it, like `@myorg/my-awesome-lib`."
    },
    "strict": {
      "type": "boolean",
      "description": "Whether to enable `tsconfig` strict mode or not.",
      "default": true
    },
    "storybookConfiguration": {
      "description": "Whether to include storybook configuration for the generated library.",
      "type": "boolean",
      "default": false
    },
    "generateComponent": {
      "description": "Whether to generate component for the library.",
      "type": "boolean",
      "default": true
    }
  },
  "required": ["name"]
}
