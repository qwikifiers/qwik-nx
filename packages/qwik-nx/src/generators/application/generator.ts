import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
