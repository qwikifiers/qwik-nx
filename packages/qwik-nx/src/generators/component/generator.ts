import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  getProjects,
  joinPathFragments,
  logger,
  names,
  Tree,
} from '@nrwl/devkit';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { ComponentGeneratorSchema } from './schema';

interface NormalizedSchema extends ComponentGeneratorSchema {
  directory: string;
  hasStyles: boolean;
  projectRoot: string;
}

function getDirectory(host: Tree, options: ComponentGeneratorSchema) {
  const workspace = getProjects(host);
  let baseDir: string;
  if (options.directory) {
    baseDir = options.directory;
  } else {
    baseDir =
      workspace.get(options.project)!.projectType === 'application'
        ? 'components'
        : 'lib';
  }
  return options.flat
    ? baseDir
    : joinPathFragments(baseDir, names(options.name).fileName);
}

function normalizeOptions(
  host: Tree,
  options: ComponentGeneratorSchema
): NormalizedSchema {
  const project = getProjects(host).get(options.project);

  if (!project) {
    logger.error(
      `Cannot find the ${options.project} project. Please double check the project name.`
    );
    throw new Error();
  }

  const projectRoot =
    project.sourceRoot ?? joinPathFragments(project.root, 'src');

  const directory = getDirectory(host, options);

  return {
    ...options,
    directory,
    hasStyles: options.style !== 'none',
    projectRoot,
  };
}

function createComponentFiles(tree: Tree, options: NormalizedSchema) {
  const libNames = names(options.name);
  const hasStyles = options.style && options.style !== 'none';
  const templateOptions = {
    ...options,
    ...libNames,
    hasStyles,
  };

  const componentDir = joinPathFragments(
    options.projectRoot,
    options.directory
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/common'),
    componentDir,
    templateOptions
  );
  if (hasStyles) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/styles'),
      componentDir,
      templateOptions
    );
  }
  if (!options.skipTests) {
    generateFiles(
      tree,
      joinPathFragments(__dirname, 'files/tests'),
      componentDir,
      templateOptions
    );
  }
}

export async function componentGenerator(
  tree: Tree,
  options: ComponentGeneratorSchema
): Promise<GeneratorCallback> {
  const normalizedOptions = normalizeOptions(tree, options);
  createComponentFiles(tree, normalizedOptions);
  await formatFiles(tree);

  return addStyledModuleDependencies(tree, normalizedOptions.style);
}

export default componentGenerator;
