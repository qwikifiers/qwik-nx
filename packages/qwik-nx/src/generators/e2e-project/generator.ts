import {
  ensurePackage,
  GeneratorCallback,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import { getInstalledNxVersion } from '../../utils/get-installed-nx-version';
import { nxKitVersion } from '../../utils/versions';
import { E2eProjectGeneratorSchema, NormalizedSchema } from './schema';
import { normalizeOptions } from './utils/normalize-options';

export async function addE2eProject(
  tree: Tree,
  options: E2eProjectGeneratorSchema
): Promise<GeneratorCallback> {
  const projectConfiguration = readProjectConfiguration(tree, options.project);

  if (projectConfiguration.projectType !== 'application') {
    throw new Error('Cannot setup e2e project for the given frontend project.');
  }

  const normalizedOptions = normalizeOptions(tree, options);
  if (options.e2eTestRunner === 'cypress') {
    return addCypress(tree, normalizedOptions);
  }

  if (options.e2eTestRunner === 'playwright') {
    return addPlaywright(tree, normalizedOptions);
  }

  return () => void 0;
}

async function addCypress(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nrwl/cypress', getInstalledNxVersion(tree));
  const { cypressProjectGenerator } = await import('@nrwl/cypress');

  return await cypressProjectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    project: options.project,
    bundler: 'vite',
    skipFormat: options.skipFormat,
  });
}

async function addPlaywright(tree: Tree, options: NormalizedSchema) {
  await ensurePackage(tree, '@nxkit/playwright', nxKitVersion);
  const { projectGenerator } = await import('@nxkit/playwright');

  return await projectGenerator(tree, {
    ...options,
    name: options.e2eProjectName,
    directory: options.directory,
    frontendProject: options.project,
    skipFormat: options.skipFormat,
  });
}

export default addE2eProject;
