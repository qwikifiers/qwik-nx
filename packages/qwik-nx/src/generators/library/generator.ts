import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { Linter } from '@nrwl/linter';
import { libraryGenerator as nxLibraryGenerator } from '@nrwl/workspace/generators';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import {
  getRelativePathToRootTsConfig,
  initGenerator as jsInitGenerator,
} from '@nrwl/js';
import { LibraryGeneratorSchema, NormalizedSchema } from './schema';
import componentGenerator from './../component/generator';
import { configureEslint } from '../../utils/configure-eslint';
import { initGenerator } from '@nrwl/vite';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';
import { getQwikLibProjectTargets } from './utils/get-qwik-lib-project-params';
import { normalizeOptions } from './utils/normalize-options';

export async function libraryGenerator(
  tree: Tree,
  schema: LibraryGeneratorSchema
) {
  const options = normalizeOptions(tree, schema);
  const tasks: GeneratorCallback[] = [];

  await jsInitGenerator(tree, {
    skipFormat: true,
  });

  tasks.push(await addLibrary(tree, options));

  tasks.push(await configureVite(tree, options));

  if (options.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, options.projectName, true));
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
  return runTasksInSerial(...tasks);
}

async function addLibrary(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tasks = [];
  const libGeneratorTask = await nxLibraryGenerator(tree, {
    name: options.name,
    directory: options.directory,
    tags: options.tags,
    linter: Linter.None,
    importPath: options.importPath,
    strict: options.strict,
    standaloneConfig: options.standaloneConfig,
    unitTestRunner: 'none',
    skipBabelrc: true,
    skipFormat: true,
  });
  tasks.push(libGeneratorTask);

  tree.delete(`${options.projectRoot}/src/lib/${options.name}.ts`);

  const templateOptions = {
    ...options,
    ...names(options.name),
    strict: !!options.strict,
    rootTsConfigPath: getRelativePathToRootTsConfig(tree, options.projectRoot),
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.projectRoot,
    templateOptions
  );

  if (!options.setupVitest) {
    tree.delete(`${options.projectRoot}/tsconfig.spec.json`);
  }

  if (!options.buildable) {
    tree.delete(`${options.projectRoot}/package.json`);
    if (!options.setupVitest) {
      tree.delete(`${options.projectRoot}/vite.config.ts`);
    }
  }

  const componentGeneratorTask = await componentGenerator(tree, {
    name: options.name,
    skipTests: !options.setupVitest,
    style: options.style,
    project: options.projectName,
    flat: true,
  });

  tasks.push(componentGeneratorTask);

  return runTasksInSerial(...tasks);
}

async function configureVite(tree: Tree, options: NormalizedSchema) {
  const callback = await initGenerator(tree, {
    uiFramework: 'none',
    includeLib: true,
  });

  const projectConfig = readProjectConfiguration(tree, options.projectName);

  projectConfig.targets = {
    ...projectConfig.targets,
    ...getQwikLibProjectTargets(options),
  };

  updateProjectConfiguration(tree, options.projectName, projectConfig);

  return callback;
}

export default libraryGenerator;
