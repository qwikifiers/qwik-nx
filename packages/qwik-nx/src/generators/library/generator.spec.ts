import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';

import generator from './generator';
import { LibraryGeneratorSchema } from './schema';

describe('library generator', () => {
  let appTree: Tree;
  const options: LibraryGeneratorSchema = { name: 'mylib' };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should run successfully', async () => {
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'mylib');

    expect(config.targets!['build']).not.toBeDefined();
    expect(config.targets!['test']).toBeDefined();
    expect(config.targets!['lint']).toBeDefined();

    expect(config).toMatchSnapshot();
    expect(
      appTree.read('libs/mylib/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      appTree.listChanges().map((c) => ({ path: c.path, type: c.type }))
    ).toMatchSnapshot();
  });

  it('should generate build configs for buildable libraries', async () => {
    await generator(appTree, { ...options, buildable: true });
    const config = readProjectConfiguration(appTree, 'mylib');

    expect(config.targets!['build']).toBeDefined();
    expect(config.targets!['test']).toBeDefined();
    expect(config.targets!['lint']).toBeDefined();

    expect(config).toMatchSnapshot();
    expect(
      appTree.read('libs/mylib/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      appTree.listChanges().map((c) => ({ path: c.path, type: c.type }))
    ).toMatchSnapshot();
  });

  it('should not generate build and test configs for non-buildable libraries w/o unit tests', async () => {
    await generator(appTree, { ...options, unitTestRunner: 'none' });
    const config = readProjectConfiguration(appTree, 'mylib');

    expect(config.targets!['build']).not.toBeDefined();
    expect(config.targets!['test']).not.toBeDefined();
    expect(config.targets!['lint']).toBeDefined();

    expect(appTree.exists('libs/mylib/vite.config.ts')).toBeFalsy();
    expect(appTree.exists('libs/mylib/tsconfig.spec.json')).toBeFalsy();

    expect(config).toMatchSnapshot();
    expect(
      appTree.listChanges().map((c) => ({ path: c.path, type: c.type }))
    ).toMatchSnapshot();
  });
});
