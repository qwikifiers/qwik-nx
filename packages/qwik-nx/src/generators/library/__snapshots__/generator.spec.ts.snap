// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library generator should generate build configs for buildable libraries 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "mylib",
  "projectType": "library",
  "root": "libs/mylib",
  "sourceRoot": "libs/mylib/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nrwl/vite:build",
      "options": Object {
        "configFile": "libs/mylib/vite.config.ts",
        "mode": "lib",
        "outputPath": "dist/libs/mylib",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mylib/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "test": Object {
      "executor": "@nrwl/vite:test",
      "options": Object {
        "passWithNoTests": true,
        "reportsDirectory": "../../coverage/libs/mylib",
      },
      "outputs": Array [
        "../..//coverage/libs/mylib",
      ],
    },
  },
}
`;

exports[`library generator should generate build configs for buildable libraries 2`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';
import { join } from 'path';

import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/libs/mylib',
  plugins: [
    qwikNxVite(),
    qwikVite(), 
    tsconfigPaths({ root: '../../' }),
      dts({
      tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
      // Faster builds by skipping tests. Set this to false to enable type checking.
      skipDiagnostics: true,
    }),
    
  ],
    mode: 'lib',
  // Configuration for building your library.
  // See: https://vitejs.dev/guide/build.html#library-mode
  build: {
    lib: {
      // Could also be a dictionary or array of multiple entry points.
      entry: 'src/index.ts',
      name: 'mylib',
      fileName: 'index',
      // Change this to the formats you want to support.
      // Don't forgot to update your package.json as well.
      formats: ['es', 'cjs'],
    },
    rollupOptions: {
      // External packages that should not be bundled into your library.
      external: [],
    },
  },
  
    test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../../coverage/libs/mylib'
    }
  },
  
});
"
`;

exports[`library generator should generate build configs for buildable libraries 3`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/index.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.lib.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.css",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintignore",
    "type": "CREATE",
  },
]
`;

exports[`library generator should not generate build and test configs for non-buildable libraries w/o unit tests 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "mylib",
  "projectType": "library",
  "root": "libs/mylib",
  "sourceRoot": "libs/mylib/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mylib/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
  },
}
`;

exports[`library generator should not generate build and test configs for non-buildable libraries w/o unit tests 2`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/index.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.lib.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.css",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintignore",
    "type": "CREATE",
  },
]
`;

exports[`library generator should run successfully 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "mylib",
  "projectType": "library",
  "root": "libs/mylib",
  "sourceRoot": "libs/mylib/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mylib/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "test": Object {
      "executor": "@nrwl/vite:test",
      "options": Object {
        "passWithNoTests": true,
        "reportsDirectory": "../../coverage/libs/mylib",
      },
      "outputs": Array [
        "../..//coverage/libs/mylib",
      ],
    },
  },
}
`;

exports[`library generator should run successfully 2`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vite';

import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/libs/mylib',
  plugins: [
    qwikNxVite(),
    qwikVite(), 
    tsconfigPaths({ root: '../../' }),
    
  ],
  
    test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../../coverage/libs/mylib'
    }
  },
  
});
"
`;

exports[`library generator should run successfully 3`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/index.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.lib.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.css",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/mylib.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintignore",
    "type": "CREATE",
  },
]
`;
